code/game/objects/items/devices/uplinks.dm

```diff
@@ -172,6 +182,18 @@ A list of items and costs is stored under the datum of every game mode, alongsid
 		show_description = text2num(href_list["show_desc"])
 		interact(usr)
 
+	else if(href_list["get_tc"])
+		if (uses <= 0)
+			return
+		var/amount = input("How many telecrystals do you wish to withdraw?:", "Extract telecrystals", null) as num
+		if (amount > uses)
```

Comment:
feedback message, also why not just snap to maximum? eg if they have 20tc and they enter 99, print out 20

Reply:
Good idea